java/src/main/java/com/logicalclocks/hsfs/engine/TrainingDatasetEngine.java

```diff
@@ -220,6 +220,10 @@ public void initPreparedStatement(TrainingDataset trainingDataset)
     // init prepared statement if it has not already
     if (trainingDataset.getPreparedStatements() == null) {
       initPreparedStatement(trainingDataset);
+      //check if primary key map correspond to serving_keys. We check this during init state to reduce latency.
+      if (!trainingDataset.getServingKeys().equals(entry.keySet())) {
+        throw new IllegalArgumentException("Provided primary key map doesn't correspond to serving_keys");
+      }
```

Comment:
This doesn't make sense, as the user can potentially provide a different set of keys at each method call.

Reply:
commented bellow but I am just checking map key names which has to be always the same